frontend/app/src/App.tsx

```diff
@@ -840,6 +841,8 @@ export class App extends PureComponent<Props, State> {
 
   handlePageProfileMsg = (pageProfile: PageProfile): void => {
     const pageProfileObj = PageProfile.toObject(pageProfile)
+
+    const browserInfo = getBrowserInfo()
```

Comment:
Since the info will not change during an app session, we could also store it into the App state and only compute in the constructor here: https://github.com/streamlit/streamlit/blob/814f1a58b78a5e47da5da532f274509c1b401b07/frontend/app/src/App.tsx#L266

But for simplicity it is probably fine to keep it here if we assume the `getBrowserInfo` call is super lightweight / efficient.

Reply:
Thanks for calling this out! I think we could even do caching at the module level then right? Essentially have a variable called sessionInfo in getBrowserInfo.ts that if it's null, we call into the library, and if it isn't we return it. Idk if that's going to mess with the unit tests, but would be even more cached if we think this is an issue (and would keep the code in App.tsx simpler!)

Reply:
maybe just run `getBrowserInfo` 10k times in a loop first and measure the average time, e.g.:

```javascript
let totalTime = 0
for (let i = 0; i < iterations; i++) {
    const startTime = performance.now()
    getBrowserInfo()
    const endTime = performance.now()
    totalTime += (endTime - startTime)
}

const averageTime = totalTime / iterations
console.log(`Average time per execution: ${averageTime} milliseconds`)
```

I assume the library only does simple string matching / maybe some light regex -> which would make it completely insignificant to optimize and we can just leave it as is.

Reply:
`Average time per execution: 0.009660000000149011 milliseconds`

seems quick enough!

Reply:
looks fine to leave as is 👍